compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -1033,8 +1026,8 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     |this| {
                         let ty = this
                             .lower_ty(ty, ImplTraitContext::Disallowed(ImplTraitPosition::ConstTy));
-                        let body = this.lower_const_body(i.span, expr.as_deref());
                         this.lower_define_opaque(hir_id, &define_opaque);
+                        let body = this.lower_anon_const_to_const_arg(body.as_deref().unwrap());
```

Comment:
this unwrap probably causes us to ice on impl assoc consts like `impl Trait for () { const ASSOC: usize; }`